diff --git a/lib/vtls/openssl.c b/lib/vtls/openssl.c
index c489600..59168ca 100644
--- a/lib/vtls/openssl.c
+++ b/lib/vtls/openssl.c
@@ -736,6 +736,11 @@ int Curl_ossl_init(void)
   /* Lets get nice error messages */
   SSL_load_error_strings();
 
+  /* Load config file */
+  OPENSSL_load_builtin_modules();
+  if (CONF_modules_load_file(getenv("OPENSSL_CONF"), NULL, 0) <= 0)
+    return 0;
+
   /* Init the global ciphers and digests */
   if(!SSLeay_add_ssl_algorithms())
     return 0;
