diff --git a/lib/easy.c b/lib/easy.c
index 160712e..0a1dc61 100644
--- a/lib/easy.c
+++ b/lib/easy.c
@@ -939,6 +939,18 @@ CURL *curl_easy_duphandle(CURL *incurl)
                              data->state.resolver) != CURLE_OK)
     goto fail;
 
+  /* If set, clone the handle to the engine being used. */
+#if defined(USE_SSLEAY) && defined(HAVE_OPENSSL_ENGINE_H)
+  if (data->state.engine) {
+    /* state.engine existing means curl_ossl_set_engine was
+     * previously successful.  Because curl_ossl_set_engine worked,
+     * we can query the already-set engine for that handle and use
+     * that to increment a reference:
+     */
+     Curl_ssl_set_engine(outcurl, ENGINE_get_id(data->state.engine));
+  }
+#endif /* USE_SSLEAY */
+
   Curl_convert_setup(outcurl);
 
   outcurl->magic = CURLEASY_MAGIC_NUMBER;
